src/main/java/de/samply/share/broker/rest/InquiryHandler.java

```diff
@@ -250,6 +230,32 @@ private int store(String query, int userid, String inquiryName,
     return returnValue;
   }
 
+  private void storeCriteria(String query, boolean isMonitoring, boolean structuredQuery,
+      Inquiry inquiry, Connection connection) {
+    if (structuredQuery) {
```

Comment:
Just switch on the query type here. Or even better: put the query into a class that has a method for saving the criteria = real Java polymorphism.